src/utils/date.ts

```diff
@@ -1,25 +1,24 @@
 import { format } from 'date-fns';
 
-export const formatDate = (
-  date: string | Date,
-  includeTime = false,
-  formatType = 'ISO8601',
-): string => {
+export const DEFAULT_DATE_FORMAT = 'yyyy/mm/dd';
+export const TIME_DATE_FORMAT = `${DEFAULT_DATE_FORMAT}Thh:mm:ss`;
+
+export const formatDate = (date: string | Date, formatType = DEFAULT_DATE_FORMAT): string => {
   if (!date) date = new Date();
   let newDate = date.toLocaleString();
   try {
     switch (formatType) {
-      case 'ISO8601':
+      case DEFAULT_DATE_FORMAT:
         newDate = format(new Date(date), 'yyyy-MM-dd');
         break;
+      case TIME_DATE_FORMAT:
+        newDate = format(new Date(date), 'yyyy-MM-dd') + ' ' + new Date(date).toLocaleTimeString();
+        break;
```

Comment:
This isn't really how I saw this working, I either get a date of `yyyy-MM-dd` returned or `yyyy-MM-ddThh:mm:ss`, what if I wanted `1st Jan 2024`, this function cannot do that, it is very specific to only 2 formats.

I was expecting just a generic format function that can handle any format I want, but default to the most common, if I don't supply a format. so, I expected something like:

```
export DEFAULT_DATE_FORMAT = 'yyyy-dd-mm';

export const formatDate =(date: Date, formatStr = DEFAULT_DATE_FORMAT) => {
  const newDate = (date typeof 'string') ? new Date(date) : date;

  return format(newDate, formatStr);
}
```

With this I can format a date (string or Date object) into any format I want. If I just want `2024-10-23` I just say `formatDate(dateObj);`, if I want time, then `formatDate(dateObj, 'yyyy-dd-mmThh:mm:ss');` If I want `23rd of Oct 2024` then `formatDate(dateObj, 'do LLL yyyy');`

This simple function means I can format a date into any format I want, and the code is agnostic to using `date-fns`. I could easily swap it out for `momentjs` or any other library, without touching code elsewhere.

I don't mind adding some extra checking that inputs are what we expect and tests can easily help here e.g. does it handle leap year Feb 29th. Is it UTC, what if I give it a British Summer Time, I want UTC not BST, if I pass an empty string, throw an error maybe. not sure if an empty string should mean using today's date, I can see that breaking things as much as helping. The code calling this function should ensure what it is passing in is correct as it will know, this function should be very generic, so can't fix wrong data inputs other than the obvious e.g. null, empty string, letters instead of proper date etc.

At least this is my opinion. Also, the date formats are not set in stone, we may find the customers don't like ISO8601 and want us to revert to local string, or they may want several formats in different places.